### ISSUE TITLE
PKL Processing Not working with Python 3.6.8

### ISSUE THREAD

**Issue by app-generator** [app-generator/devtool-python-converter#23]
PKL processing (all conversions) not working when the Python version is below 3.8. 
Future tests should use also python 3.6.8 (production version).

**Comment by app-generator** [comment]
Hello @masoom80 

The same issue with the PKL in the latest version. I'm using `Python 3.6.9`

Here is the runtime log: 

```log
127.0.0.1 - - [22/Dec/2022 06:41:31] "POST / HTTP/1.1" 500 -
Traceback (most recent call last):
  File "/home/sm0ke/work/devtool-python-converter/env/lib/python3.6/site-packages/flask/app.py", line 2091, in __call__
    return self.wsgi_app(environ, start_response)
  File "/home/sm0ke/work/devtool-python-converter/env/lib/python3.6/site-packages/flask/app.py", line 2076, in wsgi_app
    response = self.handle_exception(e)
  File "/home/sm0ke/work/devtool-python-converter/env/lib/python3.6/site-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/home/sm0ke/work/devtool-python-converter/env/lib/python3.6/site-packages/flask/app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/sm0ke/work/devtool-python-converter/env/lib/python3.6/site-packages/flask/app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/sm0ke/work/devtool-python-converter/env/lib/python3.6/site-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/home/sm0ke/work/devtool-python-converter/env/lib/python3.6/site-packages/flask/app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/sm0ke/work/devtool-python-converter/env/lib/python3.6/site-packages/flask/app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/home/sm0ke/work/devtool-python-converter/app/views.py", line 154, in index
    file1 = convert_pandas_to_csv(file)
  File "/home/sm0ke/work/devtool-python-converter/py_data_converter/converter_pandas.py", line 15, in convert_pandas_to_csv
    df = pd.read_pickle(file)
  File "/home/sm0ke/work/devtool-python-converter/env/lib/python3.6/site-packages/pandas/io/pickle.py", line 182, in read_pickle
    return pickle.load(f)
ValueError: unsupported pickle protocol: 5
```

**Comment by app-generator** [comment]
This might help:
https://stackoverflow.com/questions/63329657/python-3-7-error-unsupported-pickle-protocol-5

a solution might be to guard the load via a `try .. except` and read the file using Protocol 4 on the error as a fallback mechanism. There is a guy that suggests:

> `A concrete example of serializing to a lower version with pandas data frame: df.to_pickle('filename.pkl', protocol=4)`

Please take a look. Ty

**Comment by app-generator** [comment]
Hello guys, 

The issue is still there in the latest commits. Please check. 

> Testing ENV: Ubuntu 18.04 LTS, `Python 3.6.9`

Please check. Ty! 

### Runtime Logs: 

```bash
127.0.0.1 - - [27/Dec/2022 10:23:58] "POST / HTTP/1.1" 500 -
Traceback (most recent call last):
  File "/home/sm0ke/work/devtool-python-converter/env/lib/python3.6/site-packages/flask/app.py", line 2091, in __call__
    return self.wsgi_app(environ, start_response)
  File "/home/sm0ke/work/devtool-python-converter/env/lib/python3.6/site-packages/flask/app.py", line 2076, in wsgi_app
    response = self.handle_exception(e)
  File "/home/sm0ke/work/devtool-python-converter/env/lib/python3.6/site-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/home/sm0ke/work/devtool-python-converter/env/lib/python3.6/site-packages/flask/app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/sm0ke/work/devtool-python-converter/env/lib/python3.6/site-packages/flask/app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/sm0ke/work/devtool-python-converter/env/lib/python3.6/site-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/home/sm0ke/work/devtool-python-converter/env/lib/python3.6/site-packages/flask/app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/sm0ke/work/devtool-python-converter/env/lib/python3.6/site-packages/flask/app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/home/sm0ke/work/devtool-python-converter/app/views.py", line 118, in index
    csv_file = pd.read_pickle(file)
  File "/home/sm0ke/work/devtool-python-converter/env/lib/python3.6/site-packages/pandas/io/pickle.py", line 182, in read_pickle
    return pickle.load(f)
ValueError: unsupported pickle protocol: 5
```  

### Screen

![image](https://user-images.githubusercontent.com/51070104/209636798-3d264ea5-5d9a-4eea-843b-eacbe82c3947.png)

**Comment by masoom80** [comment]
@app-generator
Is this problem solved?

**Comment by app-generator** [comment]
Hello @masoom80 

No feature can be used or tested due to this regression:

[Latest PR - All Conversions Broken](https://github.com/app-generator/devtool-python-converter/issues/30)

I will re-test fully the app once is fixed. 

Ty!

**Comment by masoom80** [comment]
hello @app-generator 
Is this bug fixed?

**Comment by app-generator** [comment]
Hello @masoom80 

Is not working, but we can postpone the work for a few days. 

Happy new year! 

See u all in 2023. 
🚀🚀

**Comment by app-generator** [comment]
Hello @masoom80 

Tested using version `v1.0.7`. 

> STATUS: **FAILED**

In this version, the fallback to `pickle5` exits with: 

`ValueError: I/O operation on closed file.` 

Please take a look. Ty! 

*** Here are the runtime logs:

```log
127.0.0.1 - - [02/Jan/2023 10:42:59] "POST / HTTP/1.1" 500 -
Traceback (most recent call last):
  File "/home/sm0ke/work/devtool-python-converter/py_data_converter/converter_pandas.py", line 9, in convert_pandas_to_csv
    df = pd.read_pickle(file)
  File "/home/sm0ke/work/devtool-python-converter/env/lib/python3.6/site-packages/pandas/io/pickle.py", line 182, in read_pickle
    return pickle.load(f)
ValueError: unsupported pickle protocol: 5

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/sm0ke/work/devtool-python-converter/env/lib/python3.6/site-packages/flask/app.py", line 2091, in __call__
    return self.wsgi_app(environ, start_response)
  File "/home/sm0ke/work/devtool-python-converter/env/lib/python3.6/site-packages/flask/app.py", line 2076, in wsgi_app
    response = self.handle_exception(e)
  File "/home/sm0ke/work/devtool-python-converter/env/lib/python3.6/site-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/home/sm0ke/work/devtool-python-converter/env/lib/python3.6/site-packages/flask/app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/sm0ke/work/devtool-python-converter/env/lib/python3.6/site-packages/flask/app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/sm0ke/work/devtool-python-converter/env/lib/python3.6/site-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/home/sm0ke/work/devtool-python-converter/env/lib/python3.6/site-packages/flask/app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/sm0ke/work/devtool-python-converter/env/lib/python3.6/site-packages/flask/app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/home/sm0ke/work/devtool-python-converter/app/views.py", line 223, in index
    file1 = convert_pandas_to_csv(file)
  File "/home/sm0ke/work/devtool-python-converter/py_data_converter/converter_pandas.py", line 12, in convert_pandas_to_csv
    df = pickle5.load(file)
  File "/usr/lib/python3.6/tempfile.py", line 879, in read
    return self._file.read(*args)
ValueError: I/O operation on closed file.

```

**Comment by masoom80** [comment]
Hi @app-generator 
Happy new year!
I resolved the issue above but ran to a new problem. It seems like pickle5 uses pandas '_unpickle_block' attribute to read the  pkl file but pandas version 1.1.5 doesn't have this attribute.
I could not find a solution to this error.
![Screenshot 2023-01-02 181936](https://user-images.githubusercontent.com/91330997/210248090-80a40fe3-d8db-4d0a-8d1b-1511a3df1586.png)
I searched the web, and their solution was to upgrade my pandas, which is impossible in python 3.6.8.
I'll be waiting for your instruction.
TY!
